apps/full-site-editing/full-site-editing-plugin/full-site-editing-plugin.php

```diff
@@ -265,7 +265,7 @@ function populate_wp_template_data() {
 	$template_inserter->insert_default_pages();
 }
 register_activation_hook( __FILE__, __NAMESPACE__ . '\populate_wp_template_data' );
-add_action( 'after_switch_theme', __NAMESPACE__ . '\populate_wp_template_data' );
+add_action( 'switch_theme', __NAMESPACE__ . '\populate_wp_template_data' );
```

Comment:
I explored this one initially, and if I recall correctly it's running on each load, as opposed to `after_switch_theme` which runs only on actual switch theme action. Do you foresee any potential problems because of that?

Reply:
During my testing, I did not notice any odd behavior, particularly since we already check to make sure we don't do anything more than once. But I'll test it again today to make sure :)

The problem is that `after_switch_theme` does not happen immediately when you click it -- only upon the next request, which could be several seconds later. This results in the two bugs linked in the PR description.

Do you have ideas for a better solution?

Reply:
Not at the moment, I was just worried about some consequences of doing this on each load. 

> particularly since we already check to make sure we don't do anything more than once

This flag is set at the end of the process. If we get an error before that it won't be set again and the population will be attempted again on next load. That translates into a couple of more  API requests per each load (due to retries). If everything is working correctly this wont' cause any problems, but if it happens that API request to this endpoint fails for some reason (say we introduce a bug in a commit) we could end up generating too many requests and overloading the API. 🤔 (since it would be attempted on each page view)

Reply:
I just tested the behavior by adding a log statement to the very beginning of these functions before any checks happen:
1. `populate_wp_template_data` (as you say, we do not want it to run on every request)
2. `dangerously_load_full_site_editing_files` (runs on every request to load the FSE php files)

I verified that item 2 is logging as expected when visiting and editing pages, but item 1 was not logged.

So it shouldn't be running on every request and we should be safe :)